src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,854 @@
+                                                                                         RealT
+                                                                                         }
+    RealT
+end
+
+"""
+    initial_condition_weak_blast_wave(x, t, equations::IdealGlmMhdMultiIonEquations2D)
+
+A weak blast wave (adapted to multi-ion MHD) from
+- Hennemann, S., Rueda-Ramírez, A. M., Hindenlang, F. J., & Gassner, G. J. (2021). A provably entropy 
+  stable subcell shock capturing approach for high order split form DG for the compressible Euler equations. 
+  Journal of Computational Physics, 426, 109935. [arXiv: 2008.12044](https://arxiv.org/abs/2008.12044).
+  [DOI: 10.1016/j.jcp.2020.109935](https://doi.org/10.1016/j.jcp.2020.109935)
+"""
+function initial_condition_weak_blast_wave(x, t,
+                                           equations::IdealGlmMhdMultiIonEquations2D)
+    # Adapted MHD version of the weak blast wave from Hennemann & Gassner JCP paper 2020 (Sec. 6.3)
+    # Same discontinuity in the velocities but with magnetic fields
+    # Set up polar coordinates
+    inicenter = (0, 0)
+    x_norm = x[1] - inicenter[1]
+    y_norm = x[2] - inicenter[2]
+    r = sqrt(x_norm^2 + y_norm^2)
+    phi = atan(y_norm, x_norm)
+
+    # Calculate primitive variables
+    rho = zero(real(equations))
+    if r > 0.5
+        rho = 1.0
+    else
+        rho = 1.1691
+    end
+    v1 = r > 0.5 ? 0.0 : 0.1882 * cos(phi)
+    v2 = r > 0.5 ? 0.0 : 0.1882 * sin(phi)
+    p = r > 0.5 ? 1.0 : 1.245
+
+    prim = zero(MVector{nvariables(equations), real(equations)})
+    prim[1] = 1.0
+    prim[2] = 1.0
+    prim[3] = 1.0
+    for k in eachcomponent(equations)
+        set_component!(prim, k,
+                       2^(k - 1) * (1 - 2) / (1 - 2^ncomponents(equations)) * rho, v1,
+                       v2, 0, p, equations)
+    end
+
+    return prim2cons(SVector(prim), equations)
+end
+
+# 2D flux of the multi-ion GLM-MHD equations in the direction `orientation`
+@inline function flux(u, orientation::Integer,
+                      equations::IdealGlmMhdMultiIonEquations2D)
+    B1, B2, B3 = magnetic_field(u, equations)
+    psi = divergence_cleaning_field(u, equations)
+
+    v1_plus, v2_plus, v3_plus, vk1_plus, vk2_plus, vk3_plus = charge_averaged_velocities(u,
+                                                                                         equations)
+
+    mag_en = 0.5f0 * (B1^2 + B2^2 + B3^2)
+    div_clean_energy = 0.5f0 * psi^2
+
+    f = zero(MVector{nvariables(equations), eltype(u)})
+
+    if orientation == 1
+        f[1] = equations.c_h * psi
+        f[2] = v1_plus * B2 - v2_plus * B1
+        f[3] = v1_plus * B3 - v3_plus * B1
+
+        for k in eachcomponent(equations)
+            rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+            v1 = rho_v1 / rho
+            v2 = rho_v2 / rho
+            v3 = rho_v3 / rho
```

Comment:
What about that `rho_inv` business here?

Reply:
Thanks. I replaced all occurrences in ee33ce176a756cfc4571009b134dfeb841c21fde.